src/Microsoft.Health.Fhir.SqlServer.UnitTests/Features/ChangeFeed/SqlServerFhirResourceChangeDataStoreTests.cs

```diff
@@ -75,5 +77,24 @@ public async Task GivenEmptyConnectionString_WhenGetResourceChanges_ThenInvalidO
                 Assert.Equal(nameof(InvalidOperationException), ex.GetType().Name);
             }
         }
+
+        [Fact]
+        public async Task GivenOperationCanceled_WhenGetResourceChanges_ThenTaskCanceledExceptionShouldBeThrown()
+        {
+            var source = new CancellationTokenSource();
+            var token = source.Token;
+            source.Cancel();
```

Comment:
For clarification, is it safe to cancel directly here? Do we need to move this cancellation logic to line 91 after calling the GetSqlConnectionAsync function? 

I am not sure. But I'm a little concerning about that it might behave inconsistently (sometimes passes or sometimes fails) during unit test validation at PR or CI build stages. Is it fine?